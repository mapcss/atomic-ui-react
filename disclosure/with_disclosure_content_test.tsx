import WithDisclosureContent from "./with_disclosure_content.ts";
import Disclosure from "./disclosure_provider.ts";
import SSRProvider from "../ssr/ssr_provider.ts";
import {
  assertSnapshot,
  describe,
  expect,
  it,
  setupJSDOM,
} from "../dev_deps.ts";
import { render } from "@testing-library/react";

const describeTests = describe({
  name: "WithDisclosureContent",
  async beforeEach() {
    await setupJSDOM();
  },
});

it(describeTests, "should throw error when it not wrap Context", () => {
  expect(() =>
    render(
      <WithDisclosureContent>
        {(attrs) => <p {...attrs}>test</p>}
      </WithDisclosureContent>,
    )
  ).toThrow();
});

it(
  describeTests,
  "should render with style when the internal `isOpen` is false",
  (t) => {
    const { container } = render(
      <WithDisclosureContent>
        {(attrs) => <p {...attrs}>test</p>}
      </WithDisclosureContent>,
      {
        wrapper: ({ children }) => {
          return (
            <SSRProvider>
              <Disclosure>{children as never}</Disclosure>
            </SSRProvider>
          );
        },
      },
    );

    assertSnapshot(t, container.innerHTML);
  },
);

it(describeTests, "should render when the internal `isOpen` is true", (t) => {
  const { container } = render(
    <WithDisclosureContent>
      {(attrs) => <p {...attrs}>test</p>}
    </WithDisclosureContent>,
    {
      wrapper: ({ children }) => {
        return (
          <SSRProvider>
            <Disclosure initialIsOpen>{children as never}</Disclosure>
          </SSRProvider>
        );
      },
    },
  );

  assertSnapshot(t, container.innerHTML);
});
