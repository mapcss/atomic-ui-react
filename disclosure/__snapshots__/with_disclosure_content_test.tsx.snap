export const snapshot = {};

snapshot[`WithDisclosureContent > should render with style when the internal \`isOpen\` is false 1`] = `'<p id="atomic-ui-0" style="display: none;">test</p>'`;

snapshot[`WithDisclosureContent > should render when the internal \`isOpen\` is true 1`] = `'<p id="atomic-ui-0">test</p>'`;

snapshot[`WithDisclosureContent > should merge style 1`] = `'<p style="font: icon; display: none;" id="atomic-ui-0">test</p>'`;
